src/database/utilities/utilities.js

```diff
@@ -20,3 +21,44 @@ export function addBatchToParent(batch, parent, createItem) {
   if (item.batches && item.batches.find(currentBatch => currentBatch.id === batch.id)) return;
   item.addBatch(batch);
 }
+/**
+ * Removes TransactionBatch...
+ * 1 - Remove ItemBatch from TransactionBatch
+ * 2 - Remove TransactionBatch from TransactionItem
+ * 3 - if TransactionItem has no batches, remoe TransactionItem from Transaction
+ * @param {Realm}   database   App wide database
+ * @param {Transaction}   transaction  context transaction
+ * @param {TransactionBatch} transactionBatch transaction batch to remove
+ */
+export function removeTransactionBatchUtil(database, transaction, transactionBatch) {
+  // find transaction item
```

Comment:
yeah .. although it won't be called if transaction is finalised, could be good incase someone decided to use carelessly. will amend when refactoring into Transaction